tests/usecases/grant_permission_to_group_test.py

```diff
@@ -0,0 +1,198 @@
+from typing import TYPE_CHECKING
+
+from mock import ANY, call, MagicMock
+
+from grouper.constants import ARGUMENT_VALIDATION, PERMISSION_ADMIN, PERMISSION_GRANT
+from grouper.entities.permission_grant import GroupPermissionGrant
+
+if TYPE_CHECKING:
+    from tests.setup import SetupTest
+
+
+def test_permissions_grantable(setup):
+    # type: (SetupTest) -> None
+    with setup.transaction():
+        setup.add_user_to_group("gary@a.co", "some-group")
+        setup.create_service_account("service@svc.localhost", "some-group")
+        setup.grant_permission_to_group("some-permission", "arg_one", "some-group")
+        setup.grant_permission_to_group("some-permission", "arg_two", "some-group")
+        setup.grant_permission_to_group("other-permission", "*", "some-group")
+
+        # Admins should be able to grant all permissions
+        setup.add_user_to_group("zorkian@a.co", "admin-group")
+        setup.grant_permission_to_group(PERMISSION_ADMIN, "", "admin-group")
+        setup.create_service_account("admin@svc.localhost", "admin-group")
+        setup.grant_permission_to_group(PERMISSION_ADMIN, "", "admin@svc.localhost")
+
+        setup.add_user_to_group("rra@a.co", "granter-group")
+        setup.grant_permission_to_group(PERMISSION_GRANT, "some-permission", "granter-group")
+        setup.create_service_account("granter@svc.localhost", "granter-group")
+        setup.grant_permission_to_group(
+            PERMISSION_GRANT, "some-permission/arg", "granter@svc.localhost"
+        )
+        # User with no special permissions should not be able to grant anything.
+        setup.create_user("permless@a.co")
+
+    mock_ui = MagicMock()
+
+    usecase = setup.usecase_factory.create_grant_permission_to_group_usecase("rra@a.co", mock_ui)
+    expected = [
+        ("some-permission", "*"),
+    ]
+    assert usecase.permissions_grantable() == expected
+
+    all_permissions = ["some-permission", "other-permission", PERMISSION_ADMIN, PERMISSION_GRANT]
+    usecase = setup.usecase_factory.create_grant_permission_to_group_usecase(
+        "zorkian@a.co", mock_ui
+    )
+    expected = sorted([(perm, "*") for perm in all_permissions])
+    assert usecase.permissions_grantable() == expected
+
+    usecase = setup.usecase_factory.create_grant_permission_to_group_usecase("gary@a.co", mock_ui)
+    assert usecase.permissions_grantable() == []
+
+
+def test_success(setup):
```

Comment:
For completeness, could add a success case of granting where the actor is a service account